firefox-ios/Client/Frontend/Library/HistoryPanel/HistoryPanel.swift

```diff
@@ -629,6 +630,48 @@ class HistoryPanel: UIViewController,
                                      object: .selectedHistoryItem,
                                      value: .historyPanelNonGroupItem)
     }
+
+    // MARK: - LibraryPanelContextMenu
+
+    func presentContextMenu(
+        for site: Site,
+        with indexPath: IndexPath,
+        completionHandler: @escaping () -> PhotonActionSheet?
+    ) {
+        guard let contextMenu = completionHandler() else { return }
+
+        present(contextMenu, animated: true, completion: nil)
+    }
+
+    func getSiteDetails(for indexPath: IndexPath) -> Site? {
+        return siteAt(indexPath: indexPath)
+    }
+
+    func getContextMenuActions(for site: Site, with indexPath: IndexPath) -> [PhotonRowActions]? {
+        guard var actions = getDefaultContextMenuActions(
+            for: site,
+            libraryPanelDelegate: libraryPanelDelegate
+        ) else { return nil }
+
+        let removeAction = SingleActionViewModel(title: .DeleteFromHistoryContextMenuTitle,
+                                                 iconString: StandardImageIdentifiers.Large.delete,
+                                                 tapHandler: { _ in
+            self.removeHistoryItem(at: indexPath)
+        })
+
+        let pinTopSite = SingleActionViewModel(title: .AddToShortcutsActionTitle,
+                                               iconString: StandardImageIdentifiers.Large.pin,
+                                               tapHandler: { _ in
+            self.pinToTopSites(site)
+        })
+
+        actions.append(PhotonRowActions(pinTopSite))
+        actions.append(PhotonRowActions(removeAction))
+
+        let cell = tableView.cellForRow(at: indexPath)
+        actions.append(getShareAction(site: site, sourceView: cell ?? self.view, delegate: historyCoordinatorDelegate))
```

Comment:
only changed this line, but moved the code from `HistoryPanel+ContextMenuExtensions` to here